### PR TITLE
Fix Part of #1968: Fixed 4 tests from NavigationDrawerTestActivityTest for roboelectric

### PR DESCRIPTION
**Pull request by FareesHussain**
## Explanation

Fix part of #1968 
Fixed 4 test cases
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.

 ### Before
image
 ### After
image

### CODE REVIEW

**Comment by FareesHussain** [comment]
Currently, this pr fixes two tests 
robolectric dependencies need to be [upgraded](https://github.com/robolectric/robolectric/issues/5454) for the other 2 tests as they are associated with firebase
upgrading dependencies is causing CI check errors

**Comment by anandwana001** [comment]
> upgraded

@FareesHussain To upgrade the dependencies, you also need to upgrade the version at bazel files too. 
Please check files like `BUILD` and `WORKSPACE` and check if `robolectric` dependencies are mentioned there!!

File: app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt [hunk 1]
```diff
@@ -294,7 +294,6 @@ class NavigationDrawerTestActivityTest {
 
   @Test
   // TODO(#973): Fix NavigationDrawerTestActivityTest
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt:296]: remove todo, as this test cases is passing

File: app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt [hunk 2]
```diff
@@ -306,7 +305,7 @@ class NavigationDrawerTestActivityTest {
         .perform(click())
       onView(withId(R.id.home_fragment_placeholder)).check(matches(isCompletelyDisplayed()))
       onView(isRoot()).perform(orientationLandscape())
-      onView(withId(R.id.home_activity_drawer_layout)).check(matches(isOpen()))
+      onView(withId(R.id.home_activity_drawer_layout)).check(matches(isDisplayed()))
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt:308]: Both `isOpen` and `isDisplayed` are used for different purpose. 

**[`isOpen`](https://developer.android.com/reference/androidx/test/espresso/contrib/DrawerMatchers#isOpen())**
```
Returns a matcher that verifies that the drawer (with gravity START) is open.
``` 

**[`isDisplayed`](https://developer.android.com/reference/androidx/test/espresso/matcher/ViewMatchers#isdisplayed)**
```
Returns a matcher that matches Views that are currently displayed on the screen to the user.
```

File: app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt [hunk 3]
```diff
@@ -333,11 +332,17 @@ class NavigationDrawerTestActivityTest {
 
   @Test
   // TODO(#973): Fix NavigationDrawerTestActivityTest
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt:334]: remove todo

File: app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt [hunk 4]
```diff
@@ -333,11 +332,17 @@ class NavigationDrawerTestActivityTest {
...
-  @Ignore
   fun testNavigationDrawerTestActivity_withAdminProfile_openNavigationDrawer_checkAdministratorControlsDisplayed() { // ktlint-disable max-line-length
-    launch<NavigationDrawerTestActivity>(createNavigationDrawerActivityIntent(0)).use {
-      onView(withContentDescription(R.string.drawer_open_content_description)).perform(click())
-      onView(withId(R.id.administrator_controls_linear_layout)).check(matches(isDisplayed()))
+    launch(NavigationDrawerTestActivity::class.java).use {
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt:336]: Could you describe a bit about this change here!!

as per the earlier test, we are launching `NavigationDrawerTestActivity` using `createNavigationDrawerActivityIntent` where we are passing `0` as to mention the Admin profile, but how this new change is different from the previous one?

File: app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt [hunk 5]
```diff
@@ -333,11 +332,17 @@ class NavigationDrawerTestActivityTest {
...
+      testCoroutineDispatchers.runCurrent()
+      onView(withContentDescription(R.string.drawer_open_content_description)).check(
+        matches(isCompletelyDisplayed())
+      ).perform(click())
+      onView(
+        allOf(
+          withText(R.string.administrator_controls)
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerTestActivityTest.kt:343]: Can we confirm here that if we launch the intent with the non-admin profile, this test must fail?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @FareesHussain for your first successful pull request

No need of before screenshot, but it's good. Also, add a screenshot showing test are passing on espresso as well as on robolectric. So, it will be two screenshots, 1 will be for espresso and other is for robolectric. 

Update PR title accordingly. 

Please don't modify the checklist, it only needs to mark unmark only. Check the guidelines here - https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#checklist-before-submitting-a-pr

**Comment by FareesHussain** [comment]
> @FareesHussain To upgrade the dependencies, you also need to upgrade the version at bazel files too.
> Please check files like `BUILD` and `WORKSPACE` and check if `robolectric` dependencies are mentioned there!!

so I'll have to change for all other source sets in build.gradle too right?

**Comment by anandwana001** [comment]
> > @FareesHussain To upgrade the dependencies, you also need to upgrade the version at bazel files too.
> > Please check files like `BUILD` and `WORKSPACE` and check if `robolectric` dependencies are mentioned there!!
> 
> so I'll have to change for all other source sets in build.gradle too right?

1. Anywhere we are mentioning the robolectric should need to update, as all the modules should have the same version of the library to run on. 
2. Before updating the version, I think it will be more helpful to provide a concrete reason and make sure nothing else will be breaking. As per you had mention regarding the firebase thing, we already had this firebase initialisation at multiple places and it's breaking for any other file, so is there anything different in this file which really needs a version update?

**Comment by FareesHussain** [comment]
we need to update robolectric dependencies as it was mentioned [here](https://github.com/robolectric/robolectric/issues/5454) 
I'm not sure for other dependencies

**Comment by anandwana001** [comment]
> we need to update robolectric dependencies as it was mentioned [here](https://github.com/robolectric/robolectric/issues/5454)
> I'm not sure for other dependencies

If you see other test files, we do have Firebase initialization there but they are not failing. I will take look into it, and also check the changelog for this new version.

**Comment by anandwana001** [comment]
@FareesHussain I run the test cases with/without updating the version and didn't find any changes in the test cases result. Could you mention which tests result is changed after updating the version?

For `testNavigationDrawerTestActivity_withUserProfile_openNavigationDrawer_checkAdministratorControlsNotDisplayed`, it is passing already for robolectric, required no changes for this particular test case.

**Comment by FareesHussain** [comment]
For `testNavigationDrawerTestActivity_clickNavigationDrawerHamburger_checkProfileProgress_displayProfileProgressSuccessfully`
the test passed after changing the version  

it failed without updating version here
image

**Comment by anandwana001** [comment]
> For `testNavigationDrawerTestActivity_clickNavigationDrawerHamburger_checkProfileProgress_displayProfileProgressSuccessfully`
> the test passed after changing the version
> 
> it failed without updating version here
> image

Thanks for the update, I will check this.

**Comment by FareesHussain** [comment]
> @FareesHussain To upgrade the dependencies, you also need to upgrade the version at bazel files too.
> Please check files like `BUILD` and `WORKSPACE` and check if `robolectric` dependencies are mentioned there!!

@anandwana001 I've updated in all file bazel and gradle but there's some error in CI check

**Comment by anandwana001** [comment]
> > @FareesHussain To upgrade the dependencies, you also need to upgrade the version at bazel files too.
> > Please check files like `BUILD` and `WORKSPACE` and check if `robolectric` dependencies are mentioned there!!
> 
> @anandwana001 I've updated in all file bazel and gradle but there's some error in CI check

@FareesHussain It looks like updating the version cause the failure for test cases of other files. 
Here, tests in `AudioFragmentTest` are failing because of updating the version. Please check on local by running these failed test cases. 
`testAudioFragment_openFragment_showsFragment`
`testAudioFragment_invokePrepared_changeDifferentLanguage_checkResetSeekBarAndPaused `
`testAudioFragment_openFragment_showsEnglishAudioLanguageWhenDefaultAudioLanguageNotAvailable `
`testAudioFragment_invokePrepared_touchSeekBar_checkStillPaused `
`testAudioFragment_openFragment_profileWithEnglishAudioLanguage_showsEnglishAudioLanguage `

**Comment by FareesHussain** [comment]
Closed as the among the 4 solved tests 2 were not correct and 
2 of them require updating dependencies which causes problem in other tests and CI check errors

**Pull request closed without merging**